From 68f3495e243c55144240de53ae06d929edff8adb Mon Sep 17 00:00:00 2001
From: Priyanka Bose <priyanka.bose@intel.com>
Date: Thu, 14 Jul 2022 11:30:14 +0530
Subject: [PATCH] The install hook can be called for more than one module
 variant for the same library (for host). This patch modifies path len check
 so as to avoid conflict for avx2 version of libs for x86 and x86_64

Tracked-On:
Signed-off-by: Priyanka Bose <priyanka.bose@intel.com>
Change-Id: I23a485bf7d072b1ab5bd3c8aee0f58ecffa7bd43
---
 build/art.go | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/build/art.go b/build/art.go
index 79149503c510..586cb6addc71 100644
--- a/build/art.go
+++ b/build/art.go
@@ -317,7 +317,9 @@ func addTestcasesFile(ctx android.InstallHookContext) {
 	src := ctx.SrcPath().String()
 	path := strings.Split(ctx.Path().String(), "/")
 	// Keep last two parts of the install path (e.g. bin/dex2oat).
-	dst := strings.Join(path[len(path)-2:], "/")
+	//dst := strings.Join(path[len(path)-2:], "/")
+	// Keep last four parts of the install path (e.g. bin/dex2oat).
+        dst := strings.Join(path[len(path)-4:], "/")
 	if oldSrc, ok := testcasesContent[dst]; ok {
 		ctx.ModuleErrorf("Conflicting sources for %s: %s and %s", dst, oldSrc, src)
 	}
-- 
2.25.1

